File: JavaScript/4_median_of_two_sorted_arrays/solution.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    function sortNumber(a,b){
        return a - b;
    }

    var nums3 = nums1.concat(nums2).sort(sortNumber),
        m = nums1.length,
        n = nums2.length,
        flag = (m+n) % 2 === 0;
    if (flag) {
        return (nums3[(m+n)/2]+nums3[(m+n)/2-1])/2;
    } else {
        return (nums3[parseInt((m+n)/2)]);
    }
};

File: Cpp/4_median_of_two_sorted_arrays/solution.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> nums3;
        int i = 0, j = 0;
        while (i < nums1.size() || j < nums2.size())
        {
            if (i >= nums1.size()) nums3.push_back(nums2[j++]);
            else if (j >= nums2.size()) nums3.push_back(nums1[i++]);
            else if (nums1[i] < nums2[j]) nums3.push_back(nums1[i++]);
            else nums3.push_back(nums2[j++]);
        }
        if (nums3.size() % 2 == 0) return (nums3[nums3.size()/2] + nums3[nums3.size()/2 - 1])/2.0;
        else return nums3[nums3.size()/2];
    }
};